src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -340,33 +341,35 @@ protected override void OnScriptFileChanged(FileSystemEventArgs e)
             // create a TraceWriter wrapper that can be exposed to Node.js
             var log = (ScriptFunc)(p =>
             {
-                var logData = (IDictionary<string, object>)p;
-                string message = (string)logData["msg"];
-                if (message != null)
+                var logObject = (IDictionary<string, object>)p;
+                string logType = (string)logObject["type"];
+                var logData = (IDictionary<string, object>)logObject["data"];
+
+                if (logType.Equals("trace", StringComparison.OrdinalIgnoreCase))
                 {
-                    try
+                    TraceLevel level = (TraceLevel)logData["lvl"];
+                    var evt = new TraceEvent(level, (string)logData["msg"]);
+
+                    using (BeginFunctionScope(invocationContext))
                     {
-                        TraceLevel level = (TraceLevel)logData["lvl"];
-                        var evt = new TraceEvent(level, message);
-
-                        // Node captures the AsyncLocal value of the first invocation, which means that logs
-                        // are correlated incorrectly. Here we'll overwrite that value with the correct value
-                        // immediately before logging.
-                        using (invocationContext.Logger.BeginScope(
-                            new Dictionary<string, object>
-                            {
-                                ["MS_FunctionInvocationId"] = invocationContext.ExecutionContext.InvocationId
-                            }))
-                        {
-                            // TraceWriter already logs to ILogger
-                            traceWriter.Trace(evt);
-                        }
+                        // TraceWriter already logs to ILogger
+                        traceWriter.Trace(evt);
                     }
-                    catch (ObjectDisposedException)
+                }
+                else if (logType.Equals("metric", StringComparison.OrdinalIgnoreCase) &&
```

Comment:
Good feedback -- I wasn't sure the best way to add that. Check out the latest commit to see if that's what you had in mind.